### PR TITLE
Add external metrics to HPA docs

### PR DESCRIPTION
**Pull request by MaciekPytel**
Add a small section about external metrics to HPA walkthrough and mention them (as optional) in section with HPA v2 requirements.

### CODE REVIEW

**Comment by MaciekPytel** [comment]
@DirectXMan12 Can you do a technical review?

/assign @DirectXMan12

**Comment by steveperry-53** [comment]
@MaciekPytel See [this commit](https://github.com/kubernetes/website/pull/7664/commits/03b4ccd462918494c533ab55db8aafe158c18c0a) for suggested edits.

**Comment by steveperry-53** [comment]
@MaciekPytel I'm missing something, because I can't find any field in HorizontalPodAutoscaler that is related to an external metric source. For example, I don't see anything of that nature in HorizontalPodAutoscalerSpec [here](https://github.com/kubernetes/kubernetes/blob/release-1.10/staging/src/k8s.io/api/autoscaling/v1/types.go). Can you help me understand where the "type: External" yaml fits into the HorizontalPodAutoscaler manifest? Thanks.

**Comment by MaciekPytel** [comment]
@steveperry-53 It's part of v2beta1 API https://github.com/kubernetes/kubernetes/blob/release-1.10/staging/src/k8s.io/api/autoscaling/v2beta1/types.go#L113, same as custom metrics described in section above. That section starts with description of how to get HPA object to v2 representation and how different metric types work in general. I wrote the part about external metrics as a sort of subsection / extension of the section on custom metrics and so I assumed the reader would have just read about that above. Do you think I should state explicitly that the user should read the section above first?

**Comment by MaciekPytel** [comment]
@steveperry-53 I read the version your version and I fully agree with all your edits. Thanks for fixing it!

File: docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md [hunk 1]
```diff
@@ -267,6 +269,35 @@ Then, your HorizontalPodAutoscaler would attempt to ensure that each pod was con
 50% of its requested CPU, serving 1000 packets per second, and that all pods behind the main-route
 Ingress were serving a total of 10000 requests per second.
 
+### Autoscaling on metrics not related to Kubernetes objects
+
+Applications running on Kubernetes may need to autoscale based on metrics that don't have an obvious
+relationship to any object in the Kubernetes cluster, such as metrics describing a hosted service used
+by Pods. In Kubernetes 1.10 and later, you can address this use case with *external metrics*.
```
Comment (DirectXMan12) [inline docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md]: > describing a hosted service with no direct correlation to Kubernetes namespaces

File: docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md [hunk 2]
```diff
@@ -267,6 +269,35 @@ Then, your HorizontalPodAutoscaler would attempt to ensure that each pod was con
...
+section to your HorizontalPodAutoscaler manifest to specify that you need one worker per 30 outstanding tasks.
+
+```yaml
+- type: External
+  external:
+    metricName: queue_messages_ready
+    metricSelector:
+      matchLabels:
+        queue: worker_tasks
+    targetAverageValue: 30
```
Comment (DirectXMan12) [inline docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md]: describe what `targetAverageValue` does

**Comment by Bradamant3** [comment]
@MaciekPytel can you get the requested technical info changes in today, please? We need to get docs PRs merged.

File: docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md (hunk 1 above)
Comment (MaciekPytel) [reply docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md]: done

File: docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md (hunk 2 above)
Comment (MaciekPytel) [reply docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md]: Added description

**Comment by MaciekPytel** [comment]
@DirectXMan12 Addressed your comments. Can you take a look?

**Comment by Bradamant3** [comment]
@DirectXMan12 if changes are OK with you, an /lgtm from you will get this PR merged. Thanks!

**Pull request merged**